src/evt/map_api.c

```diff
@@ -351,9 +392,47 @@ ApiStatus UpdateColliderTransform(ScriptInstance* script, s32 isInitialCall) {
     return ApiStatus_DONE2;
 }
 
-INCLUDE_ASM(s32, "evt/map_api", set_zone_enabled);
+void set_zone_enabled(s32 zoneID, s32 enabled) {
+    TempSetZoneEnabled* unkStruct = &D_800D91D4[zoneID];
+
+    if (unkStruct->id1 >= 0) {
+        set_zone_enabled(unkStruct->id1, enabled);
+    }
+    if (unkStruct->id2 >= 0) {
+        set_zone_enabled(unkStruct->id2, enabled);
+    }
+
+    switch (enabled) {
+        case 0:
+            unkStruct->flags |= 0x10000;
+            break;
+        case 1:
+            unkStruct->flags &= ~0x10000;
+            break;
+    }
+}
+
+ApiStatus SetZoneEnabled(ScriptInstance *script, s32 isInitialCall) {
+    Bytecode* args = script->ptrReadPos;
+    s32 idx = get_variable(script, *args++);
+    s32 enabled = get_variable(script, *args++);
+    TempSetZoneEnabled* unkStruct = &D_800D91D4[idx];
+
+    if (unkStruct->id2 >= 0) {
+        set_zone_enabled(unkStruct->id2, enabled);
+    }
 
-INCLUDE_ASM(ApiStatus, "evt/map_api", SetZoneEnabled, ScriptInstance* script, s32 isInitialCall);
+    switch (enabled) {
+        case 0:
+            unkStruct->flags |= 0x10000;
+            break;
+        case 1:
```

Comment:
Maybe use `TRUE` and `FALSE` for these cases. Interesting they used a switch, lol